coralproject/talk
# [CORL-2376 ]quick domain moderation

## What does this PR do?
This PR implements a feature that allows moderators to create an email domain level ban at the same time they are modifying a user's ban status (in addition to the existing work flow available under Configure -> Moderation -> Email Domain

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
The `createEmailDomain` mutation has been updated to allow `MODERATOR` roles to execute it, and some additional checks have been put in place to prevent unauthorized access.

## Does this PR introduce any new environment variables or feature flags? 
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. As a moderator, site moderator, or admin, navigate to Admin -> Community
2. Choose a user who is not banned
3. Click the user's status and select "Manage Ban"
4. Observe that there is an option to "Ban all new accounts on {domain}" where domain is the domain of the user's email address
5. check this box, and ban the user (on all sites or just on some)
6. Observe that the user is banned
7. Navigate to Configure > Moderation > Email Domains and observe that the users email domain has been added to the list as "Ban all new commenter accounts"
8. This same workflow should be available from the Ban Modal when moderating reported comments (accessed via "b" hotkey
9. The options should not be available if the users email domain is already moderated
10. The option should not be available if there is no email stored for the user.
 
## How do we deploy this PR?
No special considerations should be required.
